backoff.go

```diff
@@ -11,9 +12,9 @@ type BackOff func(currentRetryCount int) time.Duration
 // ExponentialBackOff computes an exponential back-off
 func ExponentialBackOff(minTimeout time.Duration) BackOff {
 	return func(currentRetryCount int) time.Duration {
-		jitter := rand.Float64()
+		jitter := rand.Float64() + 0.5
 		strategy := 1 << currentRetryCount
-		backoff := (1 + float64(strategy)*jitter) * minTimeout.Seconds() * float64(time.Second)
+		backoff := math.Max(jitter*float64(strategy)*float64(minTimeout.Nanoseconds()), float64(minTimeout.Nanoseconds()))
```

Comment:
I think `minTimeout * strategy + jitter * X` (where `X` is some const in the range 100-300ms arbitrarily) provides better distribution of delay before the first reattempt.

With this formula for ~50% clients performing the first retry delay will be the same which can cause a spike (which goes against the purpose of jitter).